tests/ha/single_pod_restart/test_data_pod_restart_api.py

```diff
@@ -1012,3 +1015,210 @@ def test_copy_object_during_pod_restart(self):
+                    'enable_ver': True, 's3_ver': self.s3_ver}
+            resp = self.ha_api.crt_bkt_put_obj_enbl_ver(event, self.s3_test_obj, new_bucket,
+                                                        self.object_name, **args)
+            assert_utils.assert_true(resp[0], resp[1])
+            self.version_etag.update({new_bucket: []})
+            self.version_etag[new_bucket].extend(resp[1])
+            bucket_list.append(new_bucket)
+            LOGGER.info("Step 7: Created bucket and uploaded object %s of %s size. Enabled "
+                        "versioning on %s.", self.object_name, self.f_size, new_bucket)
+
+            LOGGER.info("Step 8: Upload same object %s after enabling versioning. List & verify "
+                        "the VersionID for the same for %s.", self.object_name, new_bucket)
+            args = {'file_path': download_path}
+            resp = self.ha_api.parallel_put_object(event, self.s3_test_obj, new_bucket,
+                                                   self.object_name, **args)
+            assert_utils.assert_true(resp[0], f"Upload Object failed {resp[1]}")
+            self.version_etag[new_bucket].extend(resp[1])
+            resp = self.ha_api.list_verify_version(self.s3_ver, new_bucket,
+                                                   self.version_etag[new_bucket])
+            assert_utils.assert_true(resp[0], resp[1])
+            LOGGER.info("Step 8: Uploaded same object %s after enabling versioning. Listed & "
+                        "verified the VersionID for the same for %s.", self.object_name, new_bucket)
+
+        LOGGER.info("Step 9: Get versions of %s with & without specifying VersionID & verify "
+                    "etags for %s.", self.object_name, new_bucket)
+        resp = self.ha_api.parallel_get_object(event=event, s3_ver_obj=self.s3_ver,
+                                               bkt_name=new_bucket, obj_name=self.object_name,
+                                               ver_etag=self.version_etag[new_bucket])
+        assert_utils.assert_true(resp[0], f"Get Object with versionID failed {resp[1]}")
+        resp = self.s3_test_obj.get_object(new_bucket, self.object_name)
+        latest_vetag = self.version_etag[new_bucket][-1]
+        latest_v = list(latest_vetag.keys())[0]
+        etag = list(latest_vetag.values())[0]
+        if resp[1]["VersionId"] != latest_v:
+            assert_utils.assert_true(False, "Get Object without specifying VersionID does not "
+                                            f"match with latest {latest_v} {resp[1]}")
+        if resp[1]["ETag"] != etag:
+            assert_utils.assert_true(False, "Etag without specifying VersionID does not match "
+                                            f"with etag {etag} {resp[1]}")
+        LOGGER.info("Step 9: Got versions of %s with & without specifying VersionID & verified "
+                    "etags for %s", self.object_name, new_bucket)
```

Comment:
What is the difference between step 6 and 9?

Reply:
> What is the difference between step 6 and 9?

This will go under dtm flag. It was different two different buckets but will push it under if condition.

Reply:
In case of dtm enabled, step would have done same thing twice as new_bucket = self.bucket is there.

Reply:
if dtm0_disabled is True, step 9 checks for newly created bucket and step 6 checks for self.bucket_name
if not only checks for self.bucket_name in step 6.